django/test/utils.py

```diff
@@ -841,6 +843,40 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class TimeKeeper:
+    def __init__(self):
+        self._record = []
+
+    def append(self, new_record):
+        self._record.append(new_record)
+
+    def results(self):
+        results = ""
+        for record in self._record:
+            results += "%s\n" % record
+        return results
+
+
+time_keeper = TimeKeeper()
+
+
+class time_it(TestContextDecorator):
```

Comment:
I would personally structure this a bit differently by combining `time_keeper` and `time_it`:

```python
class TimeKeeper:
    def __init__(...):
        self.records = []

   @contextlib.contextmanager
   def timed(self, name):
      t1 = time.time()
      yield
      self.records.append(...)

   def __iter__(self):
      yield from records
```

Then you can do:

```python
keeper = TimeKeeper()

with keeper.time("test"):
    ...

for line in keeper:
    print(line)
```

or something akin to that, which feels more natural.

Reply:
Yes this would be a good idea

Reply:
Yup, this is exactly what I felt I was missing when I made the initial commit.